sympy/ntheory/generate.py

```diff
@@ -836,6 +840,12 @@ def primerange(a, b=None):
         .. [1] https://en.wikipedia.org/wiki/Prime_number
         .. [2] https://primes.utm.edu/notes/gaps.html
     """
+    a = as_int(a)
+    b = as_int(b) if b is not None else None
+    yield from _primerange(a, b)
+
+def _primerange(a: int, b: int | None = None) -> Iterator[int]:
+    """ Internal implementation of primerange. """
     if b is None:
         a, b = 2, a
```

Comment:
It would be better to handle that `None` case in `primerange` so that the arguments to `_primerange` are simple.